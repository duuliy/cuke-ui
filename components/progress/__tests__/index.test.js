import React from "react";
import assert from "power-assert";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Progress from "../index";

describe("<Progress/>", () => {
  it("should render a <Progress/> components", () => {
    const wrapper = render(
      <div>
        <Progress percent={70} />
        <Progress percent={70} animation={false} />
        <Progress percent={70} circle />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render five type width <Progress/> ", () => {
    const wrapper = render(
      <div>
        <Progress percent={20} type="default" />
        <Progress percent={30} type="success" />
        <Progress percent={40} type="info" />
        <Progress percent={50} type="warning" />
        <Progress percent={60} type="error" />

        <Progress percent={20} type="default" animation={false} />
        <Progress percent={30} type="success" animation={false} />
        <Progress percent={40} type="info" animation={false} />
        <Progress percent={50} type="warning" animation={false} />
        <Progress percent={60} type="error" animation={false} />
      </div>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should find cuke-progress classnames", () => {
    const wrapper = render(
      <div>
        <Progress percent={20} />
        <Progress percent={20} animation={true} />
      </div>
    );
    assert(wrapper.find(".cuke-progress").length >= 1);
    assert(wrapper.find(".cuke-progress-enter").length >= 1);
    assert(wrapper.find(".cuke-progress-bg-animation").length === 1);
    assert(wrapper.find(".cuke-progress-bg").length >= 1);
    assert(wrapper.find(".cuke-progress-num").length >= 1);
  });

  it("should find cuke-progress-circle classnames", () => {
    const wrapper = shallow(<Progress percent={20} circle={true} />);
    assert(wrapper.find(".cuke-progress-circle").length >= 1);
    assert(wrapper.find(".cuke-progress-circle-wrapper").length === 1);
  });

  it("should render custom format", () => {
    const wrapper = shallow(
      <Progress percent={20} circle format={percent => `自定义${percent}`} />
    );
    expect(wrapper.text()).toContain("自定义20");
  });

  it("should render max and min limit [0-100]", () => {
    const wrapper = shallow(<Progress percent={20} />);
    wrapper.setProps({
      percent: 110
    });
    expect(wrapper.state().percent).toBe(100);
    wrapper.setProps({
      percent: -10
    });
    expect(wrapper.state().percent).toBe(0);
  });

  it("should update state percent when props change", () => {
    const wrapper = shallow(<Progress percent={20} />);
    wrapper.setProps({
      percent: 30
    });
    expect(wrapper.state().percent).toBe(30);
  });

  it("should can not render info when set showInfo with false", () => {
    const wrapper = render(
      <div>
        <Progress percent={20} showInfo={false} />
      </div>
    );
    assert(wrapper.find(".cuke-progress-num").length === 0);
  });
});
