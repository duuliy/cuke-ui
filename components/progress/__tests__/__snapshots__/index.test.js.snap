// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress/> should render a <Progress/> components 1`] = `
<div>
  <div
    class="cuke-progress cuke-progress-default"
    percent="70"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:70%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      70%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-default"
    percent="70"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:70%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      70%
    </div>
  </div>
  <div
    class="cuke-progress-circle-wrapper cuke-progress-default"
    percent="70"
  >
    <svg
      class="cuke-progress-circle"
      height="100"
      viewBox="0 0 100 100"
      width="100"
    >
      <circle
        class="cuke-progress-circle-bg"
        cx="50"
        cy="50"
        r="47"
      />
      <circle
        class="cuke-progress-circle-stroke-default"
        cx="50"
        cy="50"
        r="47"
        stroke-dasharray="206.7167966062084 88.59291283123218"
      />
    </svg>
    <div
      class="cuke-progress-circle-percent"
    >
      70%
    </div>
  </div>
</div>
`;

exports[`<Progress/> should render five type width <Progress/>  1`] = `
<div>
  <div
    class="cuke-progress cuke-progress-default"
    percent="20"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:20%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      20%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-success"
    percent="30"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:30%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      30%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-info"
    percent="40"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:40%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      40%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-warning"
    percent="50"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:50%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      50%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-error"
    percent="60"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:60%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      60%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-default"
    percent="20"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:20%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      20%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-success"
    percent="30"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:30%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      30%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-info"
    percent="40"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:40%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      40%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-warning"
    percent="50"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:50%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      50%
    </div>
  </div>
  <div
    class="cuke-progress cuke-progress-error"
    percent="60"
  >
    <div
      class="cuke-progress-enter"
    >
      <div
        class="cuke-progress-bg"
        style="width:60%"
      />
    </div>
    <div
      class="cuke-progress-num"
    >
      60%
    </div>
  </div>
</div>
`;
